### DISCUSSION TITLE
Compatible Wifi Adapters

### DISCUSSION THREAD

**Discussion by xchwarze** [xchwarze/wifi-pineapple-cloner#1] [Q&A] [answered]
Like all pineapple mk6 this one comes with some wifi drivers already installed.
These are: **kmod-mt76x2u**, **kmod-rt2800-usb** and **kmod-rtl8187**.
Being the only one compatible for **802.11ac/5ghz** the **kmod-mt76x2u** which can be found in the **AWUS036ACM** and the **EP-AC1605 V1**.

Please be advised that only adapters based on RT5370/MT7612U are confirmed to work correctly in all circumstances, and other WiFi adapters may not work or have available drivers at all, or may only work partially or cause unexpected behavior like crashes, reboots, etc.
Not all WiFi cards or drivers are equal in capability or performance!

More information about a specific adapter can usually found with resources such as [deviwiki](https://deviwiki.com/)

**Comment by xchwarze** [comment, marked as answer]
To help with the task of finding compatible adapters I made these listings with the data from deviwiki:

*  Based on RT3070 (rt2800 driver) [link to list](https://deviwiki.com/w/index.php?title=Special:Ask&q=%5B%5BChip1+model%3A%3A%7ERT3070%2A%5D%5D&p=format%3Dbroadtable%2Flink%3Dall%2Fheaders%3Dshow%2Fsearchlabel%3D%E2%80%A6-20further-20results%2Fclass%3Dsortable-20wikitable-20smwtable&po=%3FInterface%0A%3FForm+factor%3DFF%0A%3FInterface+connector+type%3DUSB+conn.%0A%3FFCC+ID%0A%3FManuf%0A%3FManuf+product+model%3DManuf.+mdl%0A%3FVendor+ID%0A%3FDevice+ID%0A%3FChip1+model%0A%3FSupported+802dot11+protocols%3DPHY+modes%0A%3FMIMO+config%0A%3FOUI%0A%3FEstimated+year+of+release%3DEst.+year%0A&sort=&order=&eq=no&offset=0&limit=500)
*  Based on RT5370 (rt2800 driver) [link to list](https://deviwiki.com/w/index.php?title=Special:Ask&q=%5B%5BChip1+model%3A%3ART2870%5D%5D+%5B%5BInterface%3A%3AUSB%5D%5D&p=format%3Dbroadtable%2Flink%3Dall%2Fheaders%3Dshow%2Fsearchlabel%3D%E2%80%A6-20further-20results%2Fclass%3Dsortable-20wikitable-20smwtable&po=%3FInterface%0A%3FForm+factor%3DFF%0A%3FInterface+connector+type%3DUSB+conn.%0A%3FFCC+ID%0A%3FManuf%0A%3FManuf+product+model%3DManuf.+mdl%0A%3FVendor+ID%0A%3FDevice+ID%0A%3FChip1+model%0A%3FChip2+model%0A%3FSupported+802dot11+protocols%3DPHY+modes%0A%3FMIMO+config%0A%3FOUI%0A%3FEstimated+year+of+release%3DEst.+year%0A&sort=&order=asc&eq=no&offset=0&limit=500)
*  Based on RTL8187 (rtl8187 driver) [link to list](https://deviwiki.com/w/index.php?title=Special:Ask&q=%5B%5BChip1+model%3A%3A~RTL8187*%5D%5D&p=format%3Dbroadtable%2Flink%3Dall%2Fheaders%3Dshow%2Fsearchlabel%3D%E2%80%A6-20further-20results%2Fclass%3Dsortable-20wikitable-20smwtable&po=%3FInterface%0A%3FForm+factor%3DFF%0A%3FInterface+connector+type%3DUSB+conn.%0A%3FFCC+ID%0A%3FManuf%0A%3FManuf+product+model%3DManuf.+mdl%0A%3FVendor+ID%0A%3FDevice+ID%0A%3FChip1+model%0A%3FSupported+802dot11+protocols%3DPHY+modes%0A%3FMIMO+config%0A%3FOUI%0A%3FEstimated+year+of+release%3DEst.+year%0A&sort=&order=&eq=no&offset=0&limit=500)
*  Based on MT7612U (mt76x2u driver) [link to list](https://deviwiki.com/w/index.php?title=Special:Ask&q=%5B%5BChip1+model%3A%3AMT7612U%5D%5D&p=format%3Dbroadtable%2Flink%3Dall%2Fheaders%3Dshow%2Fsearchlabel%3D%E2%80%A6-20further-20results%2Fclass%3Dsortable-20wikitable-20smwtable&po=%3FInterface%0A%3FForm+factor%3DFF%0A%3FInterface+connector+type%3DUSB+conn.%0A%3FFCC+ID%0A%3FManuf%0A%3FManuf+product+model%3DManuf.+mdl%0A%3FVendor+ID%0A%3FDevice+ID%0A%3FChip1+model%0A%3FSupported+802dot11+protocols%3DPHY+modes%0A%3FMIMO+config%0A%3FOUI%0A%3FEstimated+year+of+release%3DEst.+year%0A&sort=&order=&eq=no&offset=0&limit=500)

**Reply by novovictus** [reply]
Tested and confirmed working with an old RT3072 adapter.

**Comment by iceman3k** [comment]
Is anyone having trouble getting the USB wifi adapters to work? Specifically MT7612U based adapter? I see it using lsusb and in the interface, but the adapter is not assigned interfaces.

I ordered the brand Acaigel from newegg, https://www.newegg.com/p/2S7-08E2-00096?Item=9SIAX3GJFX6992
The chipset is listed as: MT7612U+Flash RAM

Would this really not be compatible even with the same chipset?

Works fine with Kali

**Reply by xchwarze** [reply]
I have an AWUS036ACM and another generic board with the same chipset and both work correctly.
There are some 3g routers and wifi adapters that in openwrt need to execute some extra commands to work, maybe this is your case.
Exactly happened with the adapters that when you connect them in windows let you download the driver as if it were a pendrive.

**Reply by Rekindled1977** [reply]
I just got my setup up and running using GL-AR750s. I plugged in my MT7612 and it is not showing up at all under lsusb. Any ideas why that might be?![PXL_20240104_183231656.jpg](https://github.com/xchwarze/wifi-pineapple-cloner/assets/153871975/a6c32463-0e20-44a0-8bb8-9d039aa0458b)

**Reply by xchwarze** [reply]
The driver which is the mt76x2u is installed so any problems are related to the adapter.
Usually it is either an adapter that needs to be used with a "USB mode switch" or the adapter needs to be given 1A or it won't boot.
If the adapter is of the "high power" type this usually happens. This is easily fixed by using a powered usb hub.

**Reply by Rekindled1977** [reply]
So I've been trying to figure this out and the antenna is the one that shows up as realtek on the previous image. I have tried to switch to USB mode via "usbmode -s" through ssh. No luck. I've rebooted, unplugged the antenna and plugged back in.  Do you think it's just not booting?

**Reply by xchwarze** [reply]
Attach debug log from: Help > Feedback and Support > Download Debug File

**Reply by Rekindled1977** [reply]
Here is the debug file.
[debug (2).log](https://github.com/xchwarze/wifi-pineapple-cloner/files/13837569/debug.2.log)

**Reply by Rekindled1977** [reply]
I have now tried the other recommended antenna and am having the same issues.

**Reply by xchwarze** [reply]
from your debug log:
```
Thu Jan  4 20:52:31 2024 kern.info kernel: [  604.603215] usb 1-1: new full-speed USB device number 4 using ehci-platform
Thu Jan  4 20:52:31 2024 kern.err kernel: [  604.853196] usb 1-1: device descriptor read/64, error -71
Thu Jan  4 20:52:32 2024 kern.err kernel: [  605.213169] usb 1-1: device descriptor read/64, error -71
Thu Jan  4 20:52:32 2024 kern.info kernel: [  605.703193] usb 1-1: new high-speed USB device number 5 using ehci-platform
Thu Jan  4 20:52:32 2024 kern.info kernel: [  605.917706] usb-storage 1-1:1.0: USB Mass Storage device detected
Thu Jan  4 20:52:32 2024 kern.info kernel: [  605.948443] scsi host1: usb-storage 1-1:1.0
Thu Jan  4 20:52:33 2024 kern.notice kernel: [  606.987436] scsi 1:0:0:0: CD-ROM            RTK      Driver Storage   1.00 PQ: 0 ANSI: 0 CCS
Thu Jan  4 20:52:33 2024 kern.notice kernel: [  606.996721] scsi 1:0:0:0: Attached scsi generic sg1 type 5
```

device descriptor read/64, error -71? scsi 1:0:0:0: CD-ROM ...?
clearly this device has some problem in openwrt. you should check the openwrt forums to see what they have to say about it.
you are probably experiencing this problem:
https://forum.openwrt.org/t/usbmode-not-switching-device-on-reboot-or-when-running-command-but-works-when-unplugging-comfast-cf-926ac/151353
https://github.com/openwrt/mt76/issues/393

Then, like everything else in life, the quality of the adapter is essential. The lower quality it is, the more problems you will have.
All of us who use pineapples use the same old wifi audit cards: AWUS036H, TL-WN7200ND, AWUS036ACM, etc.

**Reply by Rekindled1977** [reply]
Thanks for the feedback. I'm gonna give that a try. If that doesn't work, are you saying those other adapters you listed for the pineapple should be efficient and working?

**Reply by xchwarze** [reply]
Beyond the fact that it is openwrt, the reality is that there are a few models that work perfectly. Models that you can inject packages in Windows, macOS and Linux. Models that are really stable to work on this.
Those are the ones that never fail and are recommended whenever you want to play with WiFi.
Nowadays many of them are deprecated and that is why I published the deviwiki compatibility links.
If you need to buy a new one today you have the Alfa ones like the AWUS036ACM.

**Reply by Rekindled1977** [reply]
I am just going to order the AWUS036ACM. I'm assuming the drivers for it are there already and should plug and play?

**Reply by xchwarze** [reply]
Yes, all of those listed are already installed

**Reply by Rekindled1977** [reply]
Perfect, I just found that info reading through it as well. I'm very new to this pineapple stuff and wanted to go the diy route first in order to learn. Once I get a better understanding I'll be buying the mark 7 from hak5. I really appreciate all of your help!

**Comment by orleiv96** [comment]
Alfa USB Adapter AWUS036ACH? Chipset: Realtek RTL8812AU,  Any chance anyone knows if this works? i got  a GL slate As reccomened by yourself for £35 of ebay to make a Clone, was wondering if this would work?  or just better to go for a cheaper reccomened one many thanks

**Reply by xchwarze** [reply]
As far as I remember RTL8812AU is not supported even in openwrt 23.

**Reply by orleiv96** [reply]
Thank you for your timely response, I will order one that you’ve recommended then, hope I get this project to work! 

Honestly youre a Genius!!

**Comment by BAM00** [comment]
I saw the panda pau0b ac600 on the morrownr github adapters but its not recognizing mine
https://www.amazon.com/dp/B08NPX2X4Z/

I think this uses mt7610u chipset, anyway to get this to work?

**Reply by CriticalO** [reply]
I've had so many so called compatible WiFi adapters for so many different projects,
I purchased some cheap Chinese adapters along with two MT7612U clones, along with a few others based on supposedly supported chipsets, none of them worked. I wasted so much money that I now know the meaning of, buy cheap or buy twice, so I grabbed my self two Alfa AWUS036ACM as these support 5g as well and by the way use the MT7612U  chipset which is already installed and as soon as I plugged them in they worked perfectly I even ran bully and grabbed my neighbours Access Point Pass and Key. The Chinese clones that had the same chipset did nothing, my router did not even recognise them, more money down the pan, they were £30 for the pair another tenner and I could have had one AWUS036ACM for my troubles.

Unfortunately not all WIFi adapters are made the same, some are cheap copies with cloned chipsets, so my advice is to not buy cheap get the best adapter you can afford and you will be fine, just send that one back and swap it for the Alfa AWUS036ACM and you'll be golden, alternatively you can log into your router via SSH then try installing the driver,using the OpenWRT package manager to see if you can force it to work, something like 

opkg update
opkg install kmod-mt76x0u

To uninstall the driver to claim your wasted space back 

opkg remove kmod-mt76x0u

Note: The kmod-rt2800-usb
kmod-rt|8187and the 
kmod-mt76x2u (AWUS036ACM), are already installed, so to be honest it's one of these chipsets that you should be looking at anyway.

However if your device still doesn't work with the driver installed, you'll just have to bite the bullet and grab yourself a decent adapter, when I first started learning this stuff, everyone was banging on about the AWUS036ACH with the Realtek RTL8812AU (by the time I tried to get one they were on v2, cannot speak for v1), again, complete waste of money it is a pile of crap, just have a go at installing the drivers, that's enough to give you a headache for years, yes it can pick up signals from a long distance although you try doing anything with them and all you get is a hot adapter that's not up to it's job and does nothing of any use anyway, luckily I managed to sell it on for £37, I've had better luck with and AR9271 Clone that cost me a tenner when I shelled out £62 for that other piece of crap, the moral of this story is be very careful of what you read on the net, most of it is just copied from other sites that have no clue what they are talking about, anyway good luck...

By the way not sure why I got this message, I know nothing about anything, I wasn't there and it definitely wasn't me.
